### PR TITLE
[Quantlib]  Update port to v1.25

### PR DESCRIPTION
**Pull request by sweemer**
**Describe the pull request**

- #### What does your PR fix?  
Placeholder for QuantLib v1.25 upgrade

- #### Which triplets are supported/not supported? Have you updated the [CI baseline]
All except windows dynamic, No

- #### Does your PR follow the [maintainer guide]
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
Yes

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@sweemer ,Thanks for your pr, if this pr ready for review please let me know

**Comment by sweemer** [comment]
Sure thing, I will mark this as ready for review as soon as the upstream package is ready.

**Comment by JonLiu1993** [comment]
@sweemer ,Please run command "./vcpkg x-add-version quantlib --overwrite-version" and commit again

File: ports/quantlib/portfile.cmake [hunk 1]
```diff
@@ -11,17 +11,16 @@ if (VCPKG_TARGET_IS_WINDOWS)
     vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
 endif()
 
-vcpkg_configure_cmake(
+vcpkg_cmake_configure(
     SOURCE_PATH ${SOURCE_PATH}
```
Comment (JonLiu1993) [inline ports/quantlib/portfile.cmake:15]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/quantlib/portfile.cmake [hunk 2]
```diff
@@ -11,17 +11,16 @@ if (VCPKG_TARGET_IS_WINDOWS)
...
         -DQL_BUILD_TEST_SUITE=OFF
 )
 
-vcpkg_install_cmake()
-
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME QuantLib CONFIG_PATH lib/cmake/QuantLib)
 vcpkg_copy_pdbs()
 
 file(REMOVE_RECURSE ${CURRENT_PACKAGES_DIR}/debug/include)
```
Comment (JonLiu1993) [inline ports/quantlib/portfile.cmake:26]:
```suggestion
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
```

Comment (JonLiu1993) [reply ports/quantlib/portfile.cmake:26]: Please put double quotes around the path below
configure_file(“${SOURCE_PATH}/LICENSE.TXT” “${CURRENT_PACKAGES_DIR}/share/${PORT}/copyright” COPYONLY)

File: ports/quantlib/portfile.cmake [hunk 3]
```diff
@@ -11,17 +11,16 @@ if (VCPKG_TARGET_IS_WINDOWS)
...
         -DQL_BUILD_BENCHMARK=OFF
         -DQL_BUILD_EXAMPLES=OFF
         -DQL_BUILD_TEST_SUITE=OFF
 )
 
-vcpkg_install_cmake()
-
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(PACKAGE_NAME QuantLib CONFIG_PATH lib/cmake/QuantLib)
 vcpkg_copy_pdbs()
```
Comment (JonLiu1993) [inline ports/quantlib/portfile.cmake:25]:
```suggestion
# Install custom usage
configure_file(${CMAKE_CURRENT_LIST_DIR}/usage ${CURRENT_PACKAGES_DIR}/share/${PORT}/usage @ONLY)
```

File: ports/quantlib/portfile.cmake (hunk 2 above)
Comment (JonLiu1993) [reply ports/quantlib/portfile.cmake:26]: We should install usage in portfile.cmake

File: ports/quantlib/portfile.cmake (hunk 3 above)
Comment (JonLiu1993) [reply ports/quantlib/portfile.cmake:25]: We should install usage in portfile.cmake

**Comment by sweemer** [comment]
@JonLiu1993 Thanks for all the suggestions. I believe I applied them all - did I miss any?

**Review by JonLiu1993 — APPROVED** [review]

**Comment by strega-nil-ms** [comment]
Thanks!

**Pull request merged**